src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.HttpContentAsyncStream.cs

```diff
@@ -0,0 +1,395 @@
+                    }
+                }
+
+                VerboseTrace("TryReset failed");
+                return false;
+            }
+
+            private Task TransferAsync()
+            {
+                VerboseTrace("Initiating new CopyToAsync");
+
+                Task t;
+                try
+                {
+                    t = _content.CopyToAsync(this);
+                }
+                catch (Exception exc)
+                {
+                    t = Task.FromException(exc);
+                }
+
+                // Transfer the data from the content to this stream
+                return t.ContinueWith(p =>
+                {
+                    lock (_syncObj)
+                    {
+                        VerboseTrace("CopyToAsync completed " + p.Status + " " + p.Exception);
+
+                        // We're done transferring, but a reader doesn't know that until they successfully
+                        // read 0 bytes, which won't happen until either a) we complete an already waiting
+                        // reader or b) a new reader comes along and see _copyTask completed.  We also need
+                        // to make sure that, once we do signal completion by giving a reader 0 bytes, we're
+                        // immediately ready for a TryReset, which means the _copyTask needs to be completed.
+                        // As such, while we're holding the lock, we need to both give 0 bytes to any waiting
+                        // reader and force _copyTask to complete, which we do by replacing this Task's
+                        // reference with one known to already be complete.
+
+                        if (_asyncOp != null)
+                        {
+                            _cancellationRegistration.Dispose();
+                            Debug.Assert(_waiterIsReader, "We're done writing, so a waiter must be a reader.");
+                            if (p.IsFaulted)
+                            {
+                                _asyncOp.TrySetException(p.Exception.InnerException);
+                                _copyTask = p;
+                            }
+                            else if (p.IsCanceled)
+                            {
+                                _asyncOp.TrySetCanceled();
+                                _copyTask = p;
+                            }
+                            else
+                            {
+                                _asyncOp.TrySetResult(0);
+                                _copyTask = s_zeroTask;
+                            }
+                            Update(false, null, 0, 0, null);
+                        }
+                        else
+                        {
+                            _copyTask = p.Status == TaskStatus.RanToCompletion ? s_zeroTask : p;
+                        }
+                    }
+                }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```

Comment:
I'll fix this to pass the state into the continuation rather than using a closure.